tests/pos/i1747.scala

```diff
@@ -0,0 +1,3 @@
+abstract class Coll[E] extends java.util.Collection[E] {
+  def toArray[T](a: Array[T]): Array[T] = ???
```

Comment:
It would probably be a good idea to make this a `run` test to make sure that the erased signatures are also consistent in the bytecode.